src/Parsers/Access/ParserCreateUserQuery.cpp

```diff
@@ -141,6 +154,34 @@ namespace
             }
 
             auth_data = AuthenticationData{*type};
+            if (auth_data.getType() == AuthenticationType::SHA256_PASSWORD)
+            {
+                if (!parsed_salt.empty())
+                {
+                    auth_data.setSalt(parsed_salt);
+                }
+                else if (expect_password)
+                {
+#if USE_SSL
+                    ///generate and add salt here
+                    ///random generator FIPS complaint
+                    uint8_t key[32];
+                    RAND_bytes(key, sizeof(key));
```

Comment:
This functions returns something... according to the docs, it always returns 1. Maybe write a comment that we don't have to check the result, otherwise I stumple reading this code.

Reply:
I checked with openSSL - actually return has meaning and should be processed properly - returning 1 is just a specifics of boringssl - we shouldn't rely on such specifics
will address it in followup